tasks/config.yml

```
@@ -42,3 +42,6 @@
   set_fact:
     managed_files: "{{ managed_files |default([]) }} + \
       [ '{{ consul_configd_path }}/50custom.json' ]"
+
+- name: Validate configuration
+  shell: "{{ consul_binary }} validate -quiet {{ consul_config_path }}/config.json {{ consul_configd_path }}"
```

```suggestion
  command: >
    {{ consul_binary }} validate {{ (ansible_verbosity == 0) | ternary("-quiet", "") }}
    {{ consul_config_path }}/config.json {{ consul_configd_path }}
  changed_when: false
```

Some notes on the above suggestion, since there are several parts to it:


- CI isn't working correctly at the moment (which is something I need to look into), but ansible-lint would [complain about the use of `shell`](https://ansible.readthedocs.io/projects/lint/rules/command-instead-of-shell/).
- Same goes for [`changed-when`](https://ansible.readthedocs.io/projects/lint/rules/no-changed-when/)
- Regarding `-quiet`, the output may be useful because it also contains warnings. Let's only use `-quiet` if no `-v` arguments were given to Ansible.